### PR TITLE
Fix sticky header issue

### LINKED ISSUE #75702: FlameGraph overlaps ExploreToolbar
Since ExploreToolbar is sticky now, FlameGraphHeader overlaps it because it uses the same theme.zIndex.navbarFixed value as ExploreToolbar.

### PR DESCRIPTION
**Pull request by harisrozajac**
Making the ExploreToolbar sticky [here](https://github.com/grafana/grafana/issues/75494), has made FlameGraphHeader overlap the ExploreToolbar. The solution for this was to just pull the ExploreToolbar out of the CustomScrollbar container. This way we won't run into issues if viz plugins needed to make their components sticky. Credit for this goes to @ifrost 😄 

Before
https://github.com/grafana/grafana/assets/58232930/e29a28a6-304f-4508-928e-03ca1d928b65

After
https://github.com/grafana/grafana/assets/58232930/15c8c0e7-7544-486e-80e0-8f519221b74b

Fixes #75702

Please check that:
- [ ] It works as expected from a user's perspective.

### CODE REVIEW

**Comment by ifrost** [comment]
Oh, good catch!

In that case I wonder if instead of making the toolbar sticky, it would make more sense to just pull it out of the container and make static/fixed at the top.

I'm thinking about moving it outside of custom scrollbars:

```typescript

    return (
      <>
        <ExploreToolbar exploreId={exploreId} onChangeTime={this.onChangeTime} topOfViewRef={this.topOfViewRef} />
        <CustomScrollbar
          testId={selectors.pages.Explore.General.scrollView}
          autoHeightMin={'100%'}
          scrollRefCallback={(scrollElement) => (this.scrollElement = scrollElement || undefined)}
        >
```

Would that work while being simpler? 🤔

File: packages/grafana-flamegraph/src/FlameGraphContainer.tsx [hunk 1]
```diff
@@ -156,6 +168,8 @@ const FlameGraphContainer = ({
         colorScheme={colorScheme}
         onColorSchemeChange={setColorScheme}
         stickyHeader={Boolean(stickyHeader)}
+        stickyHeaderOffset={stickyHeaderOffset}
+        stickyHeaderZindex={stickyHeaderZindex}
```
Comment (aocenas) [inline packages/grafana-flamegraph/src/FlameGraphContainer.tsx:172]: Is the zindex change needed? I would thought the offset should prevent any overlap anyway.

Comment (harisrozajac) [reply packages/grafana-flamegraph/src/FlameGraphContainer.tsx:172]: that's a good point, i looked at and it's not needed so we can remove it 👍  i'm going to look into other ways to make ExploreToolbar stick to the top without making it sticky, based on @ifrost's comments

Comment (harisrozajac) [reply packages/grafana-flamegraph/src/FlameGraphContainer.tsx]: i've looked into this and found a better solution that doesn't require any changes in flame graph

**Comment by harisrozajac** [comment]
> Would that work while being simpler (and avoiding any changes in viz plugins)? 🤔

@ifrost yes that worked! 😸 and i've tested it with content outline and it works great and I won't have to add any offsets. thanks for your input on this!

**Comment by harisrozajac** [comment]
I have some e2e tests failing here in a couple of places but this is because apparently when cypress runs it thinks that ExploreToolbar overlays the query editor:
Screenshot 2023-09-29 at 12 00 01 PM

But in reality this NEVER happens. I'll look into this a bit more but if anyone has any idea how to fix this let me know 😄

**Comment by harisrozajac** [comment]
Turns out there was some sticky css in ExploreToolbar from my previous PR. This is now working as expected.

**Review by ifrost — APPROVED** [review]

**Pull request merged**